homework3.py

```diff
@@ -0,0 +1,86 @@
+import functools
+
+
+def divisor_of_one_hundred(func):
+    @functools.wraps(func)
+    def inner(*args, **kwargs):
+        res = func(*args, **kwargs)
+        if 100 % res == 0:
+            print("We are OK!")
+        else:
+            print(f"Bad news guys, we got {100 % res}")
+        return res
+    return inner
+
+
+def not_str(func):
+    @functools.wraps(func)
+    def inner(*args, **kwargs):
+        if True in [isinstance(el, str) for el in (args + tuple(kwargs.values()))]:
+            raise ValueError("string type is not supported")
+        return func(*args, **kwargs)
+    return inner
+
+
+def cached(func):
+    cache = {}
+    used_cache_count = 0
```

Comment:
`inner. used_cache_count = 0 `it gives you chance to call this attr when you will call your function `sum3 . used_cache_count ` and also no need then `nonlocal`